### PR TITLE
Add widget API to get dashboard properties

### PR DESCRIPTION
**Pull request by lasanthaS**
## Purpose
This PR passes the dashboard properties to its widget via widget props.

To retrieve properties from a widget:
```js
const properties = this.props.dashboard.properties;
/*
    Results:
    {
        key1: value1,
        key2: value2, 
        ...
    }
*/
```

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by lasanthaS** [comment]
> @lasanthaS It seems like this PR alters the structure of the dashboard JSON which will break existing dashboards. Please confirm.

No. This accepts both the JSON formats, but when writing it writes in the new format. So the dashboards will be backward compatible.

**Comment by this** [comment]
Old dashboar JSON structure:
```js
{
  "dashboard": {
    "url": "salesdashboard",
    "owner": "admin",
    "name": "Sales Dashboard",
    "description": "A demo dashboard which is designed using sample data.",
    "landingPage": "overview",
    "parentId": "0",
    "pages": [
      {},
      {}
    ],
    "hasOwnerPermission": false,
    "hasDesignerPermission": false,
    "hasViewerPermission": false
  },
  "widgets": {
    "generated": [],
    "custom": []
  }
}
```
The new dashboard JSON structure suggested by this PR:
```js
{
  "dashboard": {
    "url": "salesdashboard",
    "owner": "admin",
    "name": "Sales Dashboard",
    "description": "A demo dashboard which is designed using sample data.",
    "landingPage": "overview",
    "parentId": "0",
    "content": {
      "properties": [],
      "pages": [
        {},
        {}
      ]
    },
    "hasOwnerPermission": false,
    "hasDesignerPermission": false,
    "hasViewerPermission": false
  },
  "widgets": {
    "generated": [],
    "custom": []
  }
}
```
So the difference is that `pages` array is now inside the `content` property which also has the new `properties` array. 
```diff
<     "pages": [
<       {},
<       {}
<     ],
---
>     "content": {
>       "properties": [],
>       "pages": [
>         {},
>         {}
>       ]
>     },
```
@lasanthaS Can you please explain what is the reason behind moving `pages` & `properties` inside `content` object. 
IMO, we can leave the `pages` array as it is and introduce the new `properties` without any issues. Actually that approach will reduce the amount of changes we need to do.

**Comment by lasanthaS** [comment]
Yes, if that's so we have to modify the database structure in order to cater dashboard properties which will result in introducing migration scripts as well. That's why we have decided to keep the properties within the same "CONTENT" column in the dashboard database.

**Comment by mohanvive** [comment]
@lasanthaS @this Can we merge this now, please?

**Comment by this** [comment]
@lasanthaS As per the offline discussion we had, shall we pass the dashboard JSON to the `WidgetRenderer` class directly. This will eliminates the need to have a new REST endpoint to retreive dashboard properties.

**Comment by lasanthaS** [comment]
@this Yes, I tried out your suggestion and it seems like we can pass the dashboard properties via props to the widget. So we no longer require the REST endpoint and Base widget modifications. I am working on it will update the PR. Please do not merge until that.

File: components/dashboards-web-component/src/common/WidgetRenderer.jsx [hunk 1]
```diff
@@ -222,11 +222,12 @@ export default class WidgetRenderer extends Component {
             } else {
                 const widgetProps = {
                     ...this.props,
+                    dashboard: this.props.glContainer.layoutManager.dashboard,
                     width: this.props.glContainer.width,
                     height: this.props.glContainer.height,
                     pubSubHub: this.props.glContainer.layoutManager.pubSubHub,
                     muiTheme: this.state.currentTheme,
-                    channelManager: getWidgetChannelManager(),
+                    channelManager: getWidgetChannelManager()
```
Comment (this) [inline components/dashboards-web-component/src/common/WidgetRenderer.jsx]: Lets revert this change as this removal violates code styles guidelines.

**Comment by lasanthaS** [comment]
I have updated PR to pass dashboard properties to a widget via props. So you can now review and merge this.

**Comment by mohanvive** [comment]
@this I am merging this PR :smile: Please raise if you have any concerns..
